extractor/src/extract/query_builder.py

```diff
@@ -161,19 +160,27 @@ def add_attribute_to_query(self, query: Query, attribute: Attribute):
                         f"Cannot use a condition with a column that does not belong "
                         f"to the primary key table: {condition.sql_column.table}"
                     )
-                if condition.sql_column in self._condition_columns:
-                    # As we currently don't process conditions columns, we don't need to have them
-                    # duplicated in the resulting dataframe.
-                    continue
-                sqlalchemy_table = self.get_table(condition.sql_column, with_alias=False)
-                sqlalchemy_col = self.get_column(condition.sql_column, sqlalchemy_table)
-                query = query.add_columns(sqlalchemy_col)
 
-                # Add the condition columns to the _condition_columns attribute
-                self._condition_columns.add(condition.sql_column)
+                query = self.add_column_to_query(condition.sql_column, query)
 
         return query
 
+    def add_column_to_query(self, sql_column, query, sqlalchemy_table=None):
+        """ Helper function to add a column to the sqlalchemy query if it is not already present
+        and to add the column to the set of columns already present in the query.
+        """
+        if sql_column in self._cur_query_columns:
+            return query
+
+        if sqlalchemy_table is None:
+            sqlalchemy_table = self.get_table(sql_column, with_alias=False)
```

Comment:
do you think it's clearer to do this here rather than outside of this function (and make `sqlalchemy_table` mandatory) ?
I think this only happens when dealing with condition columns, so maybe it would be simpler to have:
```python
def add_attribute_to_query(self, query: Query, attribute: Attribute):
     ...
    for condition in input_group.conditions:
        ...
        sqlalchemy_table = self.get_table(sql_column, with_alias=False)
        query = self.add_column_to_query(col, query, sqlalchemy_table)

def add_column_to_query(self, sql_column, query, sqlalchemy_table):
    ...
    (remove this if)
```

Reply:
what do you think ?

Reply:
Agreed, it might be clearer